src/Speckle.Sdk/Helpers/SerializerIdWriter.cs

```diff
@@ -7,22 +9,32 @@ public sealed class SerializerIdWriter : JsonWriter
   private readonly JsonWriter _jsonWriter;
 #pragma warning disable CA2213
   private readonly JsonWriter _jsonIdWriter;
-  private readonly StringWriter _idWriter;
+  private readonly StreamWriter _idWriter;
+  private readonly MemoryStream _memoryStream;
 #pragma warning restore CA2213
 
-  public SerializerIdWriter(JsonWriter jsonWriter)
+  protected override void Dispose(bool disposing)
+  {
+    base.Dispose(disposing);
+    _idWriter.Dispose();
+    _memoryStream.Dispose();
+  }
+
+  public SerializerIdWriter(JsonWriter jsonWriter, SpeckleObjectSerializer2Pool? pool = null)
   {
     _jsonWriter = jsonWriter;
-    _idWriter = new StringWriter();
-    _jsonIdWriter = new JsonTextWriter(_idWriter);
+    _memoryStream = pool?.GetMemoryStream() ?? new MemoryStream();
+#pragma warning disable CA2000
+    _idWriter = new StreamWriter(_memoryStream);
+#pragma warning restore CA2000
+    _jsonIdWriter = pool?.GetJsonTextWriter(_idWriter) ?? new JsonTextWriter(_idWriter);
   }
 
-  public (string, JsonWriter) FinishIdWriter()
+  public string FinishIdWriter()
   {
     _jsonIdWriter.WriteEndObject();
     _jsonIdWriter.Flush();
-    var json = _idWriter.ToString();
-    return (json, _jsonWriter);
+    return Encoding.UTF8.GetString(_memoryStream.ToArray());
```

Comment:
Since the ID hashing requires a `byte[]` anyawy, perhaps we could avoid allocating this string here, and simply pass the bytes down directly?